src/main/java/gift/model/Product.java

```diff
@@ -1,4 +1,18 @@
 package gift.model;
 
-public record Product(Long id, String name, int price, String imageUrl) {
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
```

Comment:
```suggestion
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
```
위에서 말씀드린대로 build.gradle에 jakarta 의존성 추가하실 필요없구요. 이렇게 변경하시면 잘 동작할 겁니다. spring-validation의 하위 의존성으로 포함되어 있기 때문입니다.